lib/ansible/plugins/connection/paramiko_ssh.py

```diff
@@ -314,8 +359,15 @@ class Connection(ConnectionBase):
     transport = 'paramiko'
     _log_channel = None
 
+    def get_option(self, option, hostvars=None):
+        try:
+            r = super().get_option(option, hostvars=hostvars)
+        except Exception:
+            r = getattr(self._play_context, option)
```

Comment:
Is there still somethign that is relying on a play context attribute? Wouldn't it best to remove that?

Reply:
using it as a fallback, JIC someone tries to use plugin on older ansible version

Reply:
If they are copying the code then they are somewhat on the hook for any problems that occur. This should be tied to this Ansible version so I would be reluctant to have this fallback which might hide even more play context fallback problems we won't hit until later.

Reply:
That's the problem, if this is "paramiko connection now uses the correct source of data" then it shouldn't be falling back on the incorrect source of data. We ship this with Ansible and I don't see us at any point putting it in a collection. Regardless the option model shouldn't be tied to an Ansible version but rather that's in the plugins documentation block. I just see having this fallback as a code smell, we now have this code that people don't know if it's needed or not or whether it's left over from something in the past. By removing the fallback we might encounter some issues in the future but at least we can see that the plugin is no longer dependent on play context altogether.